app/src/test/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModelTest.kt

```diff
@@ -23,28 +20,22 @@ class PartiesViewModelTest : ShouldSpec(
 	{
 		extension(CoroutineTestExtension())
 
-		val partiesRemote: List<PartyRemote> = mockk()
-		val parties: List<Party> = mockk()
-		val repository: PartiesRepository = mockk()
-		lateinit var tested: PartiesViewModel
-		beforeSpec {
+		should("set state WHEN initialized according to fetch parties result") {
+			val partiesRemote: List<PartyRemote> = mockk()
+			val parties: List<Party> = mockk()
 			mockkStatic(List<PartyRemote>::toUI)
 			every { partiesRemote.toUI() } returns parties
-		}
-
-		afterSpec {
-			unmockkAll()
-		}
-
-		should("set state WHEN initialized according to fetch parties result") {
-			listOf(
-				emptyFlow<List<PartyRemote>>() to Loading,
-				flowOf(partiesRemote) to Loaded(parties),
-				flow<List<PartyRemote>> { throw Exception() } to Error,
-			).forAll { (result, state) ->
-				every { repository.parties() } returns result
+			testState<Flow<List<PartyRemote>>, List<Party>>(
+				loadingAnswer = { returns(emptyFlow()) },
+				errorAnswer = { returns(flow { throw Exception() }) },
+				loadedAnswer = { returns(flowOf(partiesRemote)) },
+				loadedData = parties,
+			) { (state, answers) ->
```

Comment:
The idea seems to be interesting in general but I feel that it's too sophisticated at first glance 😄 
At this moment I would stay with previous approach of using `mockkAnswer` and adapt it to the remaining tests. 
It seems to be more complicated to use `testState` and in the end I don't see much simplification in comparison to the previous approach, even in terms of needed lines of code. 
Let's wait for others opinion and in the meantime I would think about more deeply 😄 

Anyway, big shout-out for such proposals!

Reply:
Cuz it's seems the pr is ready to merge but not all of us are convinced to this simplification, so maybe let's vote and either merge or close this pr? @michal-kucznerowicz  @MarekMacko @BorysChajdas @kmaslowiec @marcinWisnia

Reply:
Nice initiative @bartek977, thanks :)

Reply:
Merge 👍
Close 👎
@MarekMacko @bartek977 @BorysChajdas @kmaslowiec @marcinWisnia

Reply:
Closing PR based on voting result